index.js

```diff
@@ -1,68 +1,155 @@
 'use strict';
 const EventEmitter = require('events');
-const path = require('path');
+const {
+	basename,
+	extname,
+	join,
+	relative,
+	sep,
+	isAbsolute,
+	dirname
+} = require('path');
 const os = require('os');
 const pMap = require('p-map');
 const arrify = require('arrify');
-const globby = require('globby');
-const hasGlob = require('has-glob');
 const cpFile = require('cp-file');
-const junk = require('junk');
 const pFilter = require('p-filter');
 const CpyError = require('./cpy-error');
-
+const GlobPattern = require('./glob-pattern');
+const glob = require('globby');
+
+/**
+ * @typedef {object} Options
+ * @property {number} concurrency
+ * @property {boolean} ignoreJunk
+ * @property {boolean} flat
+ * @property {string} cwd
+ * @property {boolean} overwrite
+ * @property {string|Function} rename
+ */
+
+/**
+ * @typedef {object} CopyStatus
+ * @property {number} written
+ * @property {number} percent
+ */
```

Comment:
I don't really see the point of these JSDoc definitions when we have a TypeScript definition.

Reply:
Right, I've switched to `@type {import('./index').Options}` and moved CopyStatus to index.d.ts. I'm leaving other JSDoc comments - those make sense in non TS file